src/_pytest/logging.py

```diff
@@ -447,7 +447,12 @@ def clear(self) -> None:
         self.handler.reset()
 
     def set_level(self, level: Union[int, str], logger: Optional[str] = None) -> None:
-        """Set the level of a logger for the duration of a test.
+        """Sets the threshold for this logger to level.
+
+        Logging messages which are less severe than level will be ignored; logging messages
+        which have severity level or higher will be emitted by whichever handler or handlers
+        service this logger, unless a handler’s level has been set to a higher severity level
+        than `level`.
```

Comment:
I'm not sure some of these details from the logging module make sense in the context of caplog. My suggestion is keep it short and simple 😃 
```suggestion
        Logging messages which are less severe than this level will not be captured.
```

Reply:
Agree, good points. @Vijay-Arora could you implement the change plase?

Reply:
Hi @nicoddemus and @MetRonnie , Thanks for the feedback; I'll definitely make the changes.